Library/Homebrew/bintray.rb

```diff
@@ -30,8 +30,10 @@ def initialize(org: "homebrew")
     ENV["HOMEBREW_FORCE_HOMEBREW_ON_LINUX"] = "1" if @bintray_org == "homebrew" && !OS.mac?
   end
 
-  def open_api(url, *extra_curl_args, auth: true)
-    args = extra_curl_args
+  sig { params(url: String, extra_curl_args: T::Array[String], auth: T::Boolean).returns(SystemCommand::Result) }
+  def open_api(url, extra_curl_args, auth: true)
```

Comment:
You don't actually have to rewrite `extra_curl_args` to be an array. Only when passing splats you have to use `T.unsafe`.

Maybe we should disable type-checking for splats for now, given we use them a lot and they are a first-class Ruby feature.

The error is “Sorbet does not have great support for splats right now.”, so hopefully it will have better support in future.

Reply:
I've restored the splat functionality of `open_api`, and it checks with `strict` if I make the curl call `unsafe`, but I weirdly get a single instance of a call to `open_api` reporting type problems, even if it's wrapped in unsafe. The changes I just pushed double-wrap unsafe which gets it to type check but this seems like an actual bug in Sorbet rather than an unimplemented type thing.

```
% brew typecheck
bintray.rb:72: Splats are only supported where the size of the array is known statically https://srb.help/7019
    72 |    result = open_api(url, *args)
                     ^^^^^^^^^^^^^^^^^^^^
% brew typecheck
bintray.rb:72: Splats are only supported where the size of the array is known statically https://srb.help/7019
    72 |    result = open_api(url, *T.unsafe(args))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```